### PR TITLE
[codex] fix agent edit whitespace matching

### PR DESCRIPTION
**Pull request by soluna**
## What changed

- Wrap the local Eino filesystem backend with a narrow `edit_file` fallback that only activates after the original exact edit reports `string not found`.
- The fallback trims trailing spaces and tabs per line for matching, then edits only when the normalized `old_string` maps to exactly one original file span.
- Keep shell execution on the original local backend and leave all other filesystem operations unchanged.
- Add focused tests for successful trailing-whitespace recovery, ambiguous normalized matches, and non-matching content that must still fail.
- Document the fix in `CHANGELOG.md`.

## Why

Nova currently passes model-generated `old_string` values directly to Eino local backend editing, which matches literal bytes including whitespace. If the model drops invisible trailing spaces or tabs from a snippet it just read, `edit_file` fails even though the intended replacement is otherwise unambiguous.

This keeps the recovery conservative: it does not do prefix matching, similarity matching, or bulk fuzzy replacement. If normalization still leaves zero or multiple matches, the edit fails instead of guessing.

## Validation

- `go mod tidy`
- `go test ./internal/agent`
- `go test ./...`
- `./build.sh`

### CODE REVIEW

**Comment by soluna** [comment]
Hi @alfredxw, thanks for maintaining Nova. I’m contributing from the fork because the current account does not have write access to this repository. If you expect more follow-up fixes from this account, would you be open to inviting @soluna as a collaborator with write access? No worries if you prefer the fork + PR workflow; I can continue sending focused PRs this way.

**Comment by soluna** [comment]
我能帮忙改 bug 吗

**Comment by alfredxw** [comment]
喜哥🐮比

File: internal/agent/whitespace_edit_backend.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/cloudwego/eino/adk/filesystem"
+)
+
+// whitespaceEditBackend makes edit_file tolerant to trailing spaces and tabs
```
Comment (alfredxw) [inline internal/agent/whitespace_edit_backend.go]: 感觉叫 whitespaceEditBackend 太特化了，定位成一个统一的 fs backend wrapper 吧，改下名字
